patches/master_patch.patch

```diff
@@ -488,13 +488,45 @@ index 01d3ee8917c2d05943e42556a4221161606c451d..f801248b1017abaf706533ff0889295c
 +  g_browser_process->GetTabManager()->DestroyOldContents(
 +      std::move(old_contents_deleter));
  
-   LifecycleState previous_state = GetState();
-   SetState(LifecycleState::DISCARDED);
+   SetState(LifecycleUnitState::DISCARDED,
+            DiscardReasonToStateChangeReason(discard_reason));
+@@ -932,6 +936,7 @@ void TabLifecycleUnitSource::TabLifecycleUnit::CheckIfTabIsUsedInBackground(
+                                                       decision_details);
+   }
+ 
++/*
+   // Do not freeze/discard a tab that has active WebUSB connections.
+   if (auto* usb_tab_helper = UsbTabHelper::FromWebContents(web_contents)) {
+     if (usb_tab_helper->IsDeviceConnected()) {
+@@ -939,6 +944,7 @@ void TabLifecycleUnitSource::TabLifecycleUnit::CheckIfTabIsUsedInBackground(
+           DecisionFailureReason::LIVE_STATE_USING_WEB_USB);
+     }
+   }
++*/
+ 
+   // Do not freeze tabs that are currently using DevTools.
+   if (DevToolsWindow::GetInstanceForInspectedWebContents(web_contents)) {
+diff --git a/chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc b/chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc
+index e936f25145075e6882537d6557f5dd3199d225fa..5196f97e858a191e87f1bdf66eb1c6a48a18fefe 100644
+--- a/chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc
++++ b/chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc
+@@ -296,8 +296,9 @@ void TabLifecylesEnterprisePreferenceMonitor::GetPref() {
+   // lifecycles feature.
+   const PrefService::Preference* pref =
+       pref_service_->FindPreference(prefs::kTabLifecyclesEnabled);
+-  if (pref->IsManaged() && !pref->GetValue()->GetBool())
+-    enabled = false;
++/*  if (pref->IsManaged() && !pref->GetValue()->GetBool())
```

Comment:
what is this change for?

Reply:
It was a hack that was pending to fix, because there is a crash happening on Linux startup due to that check:
```
#0  0x00000000048d4dc3 in PrefService::Preference::IsManaged() const (this=0x0) at ../../components/prefs/pref_service.cc:595
#1  0x0000000003830638 in resource_coordinator::TabLifecylesEnterprisePreferenceMonitor::GetPref() (this=0x4ed81bed6a0)
    at ../../chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc:299
#2  0x000000000382ff98 in resource_coordinator::TabLifecylesEnterprisePreferenceMonitor::TabLifecylesEnterprisePreferenceMonitor(PrefService*, base::RepeatingCallback<void (bool)>) (this=0x4ed81bed6a0, pref_service=<optimized out>, callback=...) at ../../chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc:286
#3  0x000000000382ff98 in std::__1::make_unique<resource_coordinator::TabLifecylesEnterprisePreferenceMonitor, PrefService*, base::RepeatingCallback<void (bool)> >(PrefService*&&, base::RepeatingCallback<void (bool)>&&) (__args=<optimized out>, __args=...) at ../../buildtools/third_party/libc++/trunk/include/memory:3114
#4  0x000000000382ff98 in resource_coordinator::TabLifecycleUnitSource::OnFirstLifecycleUnitCreated() (this=0x7fffffff9760)
    at ../../chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc:116
#5  0x000000000382dd94 in resource_coordinator::TabLifecycleUnitSource::TabLifecycleUnit::TabLifecycleUnit(resource_coordinator::TabLifecycleUnitSource*, base::ObserverList<resource_coordinator::TabLifecycleObserver, false, true>*, resource_coordinator::UsageClock*, content::WebContents*, TabStripModel*) (this=
    0x4ed811dc900, source=<optimized out>, observers=0x4ed81161730, usage_clock=0x9ddfea08eb382d69, web_contents=0x4ed81e54600, tab_strip_model=0x4ed81d12aa0)
    at ../../chrome/browser/resource_coordinator/tab_lifecycle_unit.cc:249
#6  0x0000000003830220 in std::__1::make_unique<resource_coordinator::TabLifecycleUnitSource::TabLifecycleUnit, resource_coordinator::TabLifecycleUnitSource*, base::ObserverList<resource_coordinator::TabLifecycleObserver, false, true>*, resource_coordinator::UsageClock* const&, content::WebContents*&, TabStripModel*&>(resource_coordinator::TabLifecycleUnitSource*&&, base::ObserverList<resource_coordinator::TabLifecycleObserver, false, true>*&&, resource_coordinator::UsageClock* const&, content::WebContents*&, TabStripModel*&) (__args=@0x4ed81161760: 0x4ed811cbaf0, __args=<optimized out>, __args=<optimized out>, __args=<optimized out>, __args=<optimized out>)
    at ../../buildtools/third_party/libc++/trunk/include/memory:3114
#7  0x0000000003830220 in resource_coordinator::TabLifecycleUnitSource::TabInsertedAt(TabStripModel*, content::WebContents*, int, bool) (this=0x4ed81161690, tab_strip_model=0x4ed81d12aa0, contents=0x4ed81e54600, index=<optimized out>, foreground=<optimized out>) at ../../chrome/browser/resource_coordinator/tab_lifecycle_unit_source.cc:186
#8  0x000000000383a5af in TabStripModel::InsertWebContentsAt(int, std::__1::unique_ptr<content::WebContents, std::__1::default_delete<content::WebContents> >, int) (this=<optimized out>, index=0, contents=..., add_types=9) at ../../chrome/browser/ui/tabs/tab_strip_model.cc:351
#9  0x000000000383e8c9 in TabStripModel::AddWebContents(std::__1::unique_ptr<content::WebContents, std::__1::default_delete<content::WebContents> >, int, ui::PageTransition, int) (this=0x4ed81d12aa0, contents=..., index=0, transition=ui::PAGE_TRANSITION_TYPED, add_types=1) at ../../chrome/browser/ui/tabs/tab_strip_model.cc:853
#10 0x00000000038de011 in extensions::TabHelper::SetBrowser(Browser*) (this=0x4ed81dea600, browser=<optimized out>)
    at ../../electron/atom/browser/extensions/tab_helper.cc:453
#11 0x000000000388e653 in atom::api::WebContents::AddNewContents(content::WebContents*, std::__1::unique_ptr<content::WebContents, std::__1::default_delete<content::WebContents> >, WindowOpenDisposition, gfx::Rect const&, bool, bool*) (this=<optimized out>, source=0x4ed81cbb900, new_contents_unique=..., disposition=WindowOpenDisposition::NEW_FOREGROUND_TAB, initial_rect=..., user_gesture=true, was_blocked=0x7fffffff9bc0) at ../../electron/atom/browser/api/atom_api_web_contents.cc:841
#12 0x0000000003896bc7 in atom::api::WebContents::OnTabCreated(mate::Dictionary const&, base::OnceCallback<void (content::WebContents*)>, content::WebContents*) (this=<optimized out>, options=..., callback=..., tab=0x4ed81e54600) at ../../electron/atom/browser/api/atom_api_web_contents.cc:2946
#13 0x000000000389ef78 in base::internal::FunctorTraits<void (atom::api::WebContents::*)(mate::Dictionary const&, base::OnceCallback<void (content::WebContents*)>, content::WebContents*), void>::Invoke<void (atom::api::WebContents::*)(mate::Dictionary const&, base::OnceCallback<void (content::WebContents*)>, content::WebContents*), atom::api::WebContents*, mate::Dictionary, base::OnceCallback<void (content::WebContents*)>, content::WebContents*>(void (atom::api::WebContents::*)(mate::Dictionary const&, base::OnceCallback<void (content::WebContents*)>, content::WebContents*), atom::api::WebContents*&&, mate::Dictionary&&, base::OnceCallback<void (content::WebContents*)>&&, content::WebContents*&&) (method=<optimized out>, args=..., args=..., receiver_ptr=<optimized out>, args=<optimized out>) at ../../base/bind_internal.h:507
#14 0x000000000389ef78 in base::internal::InvokeHelper<false, void>::MakeItSo<void (atom::api::WebContents::*)(mate::Dictionary const&, base::OnceCallback<void (content::WebContents*)>, content::WebContents*), atom::api::WebContents*, mate::Dictionary, base::OnceCallback<void (content::WebContents*)>, content::WebContents*>(void (atom::api::WebContents::*&&)(mate::Dictionary const&, base::OnceCallback<void (content::WebContents*)>, content::WebContents*), atom::api::WebContents*&&, mate::Dictionary&&, base::OnceCallback<void (content::WebContents*)>&&, content::WebContents*&&) (functor=<optimized out>, args=..., args=..., args=<optimized out>, args=<optimized out>)
    at ../../base/bind_internal.h:607
```